app/javascript/app/document-capture/components/file-input.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import FileImage from './file-image';
+import useInstanceId from '../hooks/use-instance-id';
+
+/**
+ * Returns true if the given file object is an image, or false otherwise.
+ *
+ * @param {File} file File object to test.
+ *
+ * @return {boolean} Whether given file is an image.
+ */
+export function isImageFile(file) {
+  return /^image\//.test(file.type);
+}
+
+function FileInput({ label, hint, accept, value, onChange }) {
+  const instanceId = useInstanceId();
+  const inputId = `file-input-${instanceId}`;
+  const hintId = `${inputId}-hint`;
+
+  return (
+    <>
+      {/*
+       * Disable reason: The Airbnb configuration of the `jsx-a11y` rule is strict in that it
```

Comment:
Is it possible to override the airbnb config for this? It seems like a lot to have to drop an explanatory comment like this every time we make a form label

Reply:
> Is it possible to override the airbnb config for this? It seems like a lot to have to drop an explanatory comment like this every time we make a form label

Yes, it would be possible. I hadn't yet come to some conclusion about the rule, especially in mind of the discussions at https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/718 and https://github.com/airbnb/javascript/pull/2136. It appears there _may_ be some potential conflict where this rule is justified, though it specifically causes some hardship for these sorts of hint texts (related comment at https://github.com/airbnb/javascript/pull/2136#issuecomment-643483699). I'd considered if it might be best to raise this for further discussion with the USWDS team and whether this sort of hint text markup is more beneficial than the potential risk for conflicts with assistive technology.

But also worth noting that in the broader case, we can probably abide by the rule and use both `htmlFor` and `label` nesting. It's just in these specific instances of "Label + Hint + Input" where the rule must be overridden.

Reply:
As a result of some follow-up discussion, it seems that the USWDS code should be sufficiently accessible. While there is some potential concern about incompatibilities that led Airbnb to force _both_ methods of associating labels and inputs, there is no evidence to support that this is the case, and the use-case of input hints appears to be both legitimate and incompatible with label nesting.

In the general case, we could still consider to try to use both label nesting and `for` association, though it's not clear yet if it'd be best to just disable this rule outright. I'll defer that to a later decision.